### PR TITLE
Avoid syncing blocks up and down in UI

### PR DESCRIPTION
**Pull request by juanky201271**

### CODE REVIEW

**Review by zancas — DISMISSED** [review]

**Comment by zancas** [comment]
Do we need to think more carefully about how this case occurs?

**Comment by juanky201271** [comment]
> Do we need to think more carefully about how this case occurs?

I noticed about this only in old wallets doing a long sync, and only in some group of batches... is not general.

File: app/rpc/RPC.ts [hunk 1]
```diff
@@ -882,6 +882,13 @@ export default class RPC {
         //  syncProcessStalled,
         //);
 
+        // if current block is lower than the previous current block
```
Comment (zancas) [inline app/rpc/RPC.ts:885]: This could happen during a reorg..   is this the right behavior in that case?

Comment (juanky201271) [reply app/rpc/RPC.ts:885]: How the sync process behave during the reorg? (I don't know) For me there is no point when the current block is going down. In all my test always the current block is up all the time. The only time I noticed something wrong is with the witnesses... and was with old wallets,

**Comment by juanky201271** [comment]
@zancas ready for review...

File: app/rpc/RPC.ts (hunk 1 above)
Comment (AloeareV) [reply app/rpc/RPC.ts:885]: I think the witnesses updated info can go backwards when a new note is received, as we have another witness to update...so there's more work to be done than there was before we knew about the note. I'm not sure that's what's happening, though...

Comment (AloeareV) [reply app/rpc/RPC.ts:885]: Hang on, how is 'current block' defined? Is it based off of the blocks that are completely processed? As in, fetched, trial_decrypted, and witness_updated?

Comment (juanky201271) [reply app/rpc/RPC.ts:885]: with the info from `syncstatus` first the App do this:
`const progress_blocks: number = (synced_blocks + trial_decryptions_blocks + witnesses_updated) / 3;`

after that the App do this:
`let current_block = end_block + progress_blocks;`

And then the App use `current_block` only for visualization (UI). I think is better that the use see the current block going always up. The refresh is 5 seconds, normally in 5 seconds the sync process synced a fair amount of blocks, and this is the reason why I add 1 fake block every 5 seconds only in the case that the current block is going down. I don't know if I explain myself properly...

Comment (zancas) [reply app/rpc/RPC.ts:885]: OK..   I think that the point that `current_block` is only for display means I am willing to merge.   But I am going to create an issue to track down this unexpected behavior.

**Review by zancas — APPROVED** [review]

**Pull request merged**